src/app/shared/auth/auth.service.ts

```diff
@@ -29,182 +21,103 @@ export enum AuthServiceState { Idle, Fetching, Refreshing, Error }
-  private tokenRefreshSubscription?: Subscription;
+  status$ = new BehaviorSubject<AuthStatus>(notAuthenticated());
+  failure$ = new Subject<void>();
 
   constructor(
     private oauthService: OAuthService,
-    private tempAuth: TempAuthService,
     private authHttp: AuthHttpService
   ) {
-    logState('Init the auth service');
-
-    this.state$.next(AuthServiceState.Fetching); // will immediately be changed if we can get a token without fetching
-
-    this.subscription = this.accessToken.pipe(pairwise()).subscribe(tokens => this.tokenChanged(tokens));
-
-    // First, check if a code/state is given in URL (i.e., we are back from a oauth login redirect) and try to get a token from it.
+    // (1) check if a code/state is given in URL (i.e., we are back from a oauth login redirect) and try to get a token from it.
     oauthService.tryCompletingCodeFlowLogin().pipe(
-      catchError(_e => of<AccessToken|null>(null)),
-      switchMap(newToken => {
-        const fromStorage = AccessToken.fromStorage();
-
-        // If can get a token from these code/state, use this one in priority
-        if (newToken !== null) {
-          // if there was a stored token, revoke the token
-          if (fromStorage === null) return of(newToken);
-          else return merge(
-            of(newToken),
-            this.authHttp.revokeToken(fromStorage.accessToken).pipe(
-              switchMap(() => of<AccessToken>()) // just completing without emitting anything new
-            )
-          );
+      catchError(_e => {
+        // (2) use the ongoing authentication if any
+        if (appConfig.useTokens) {
+          const token = tokenAuthFromStorage();
+          return token ? of(token) : throwError(new Error('no token stored for token auth'));
+        } else {
+          return this.authHttp.refreshCookie(); // will fail if the browser has no cookie
         }
-
-        // if there is a valid token in storage, use it
-        if (fromStorage !== null) return of(fromStorage);
-
-        // otherwise, create a temp session
-        return this.tempAuth.login().pipe(retry(2));
       }),
+      catchError(_e =>
+        // (3) otherwise, create a temp session
+        this.authHttp.createTempUser().pipe(retry(2))
+      ),
     ).subscribe({
-      next: (token: AccessToken) => {
-        this.accessToken.next(token);
+      next: (auth: AuthResult) => {
+        this.status$.next(auth);
       },
       error: _e => {
         // if temp user creation fails, there is not much we can do
-        this.state$.next(AuthServiceState.Error);
-      },
-      complete: () => {
-        // in any of the success case
-        this.state$.next(AuthServiceState.Idle);
+        this.failure$.next();
       }
     });
-  }
 
-  ngOnDestroy(): void {
-    this.subscription.unsubscribe();
-    this.accessToken.complete();
+    // setup auto-refresh after delay
+    this.status$.pipe(
+      switchMap(auth => {
+        const maxDelay = 2147483647;
```

Comment:
What does `2147483647` correspond to ? would it make sense to declare it using `x * SECONDS` or `x * MINUTES` ?

Reply:
That's not really new to this PR.  But the explanation comment which was going with this line has been lost in the transformation.
That 's not really second, minutes otherwise it would be rounded. Actually this looks like 2^31-1 after which there must be an overflow somewhere.
I'll re-add this missing explanation.